python/build_definitions/cassandra_cpp_driver.py

```diff
@@ -69,4 +69,9 @@ def get_additional_cxx_flags(self, builder: 'BuilderInterface') -> List[str]:
             # Needed to avoid this error:
             # https://gist.githubusercontent.com/mbautin/d1ce54c995f9e535ab214a12945d2e7b/raw
             extra_cxx_flags.append('-Wno-error=free-nonheap-object')
+            if gcc_major_version >= 12:
```

Comment:
Do we need to check version? Do the old compilers complain when passed an unknown flags? If not, we can always add the flags

Same for `llvm_major_version >= 16`

Reply:
Yes, old compilers complain if passed an unknown flag. And I would like to keep the behavior for old compilers the same. But we can remove support for compilers older than Clang 14 now.

Reply:
Or, since this is GCC, I guess, older than GCC 11? Anyway, I prefer to keep the logic for old compiler versions the same to avoid having to re-test if the build still works for old compilers.